### PR TITLE
Eoin/UI component health

### PR DESCRIPTION
**Pull request by boilsquid**
# Why
This PR adds the ability to render a Form constructed in the Evervault UI, within UI Components

# How
- Add a new `form` component that will render out a form pulled from the Evervault UI
- Add appropriate styling for default theming

### CODE REVIEW

File: packages/ui-components/src/Form/index.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  return (
+    <div>
+      <form id={config.formUuid} onSubmit={(event) => { void handleSubmit(event); }}>
+        {formElements?.map((element) => {
+          const renderField = fieldRenderers[element.elementType];
+          return renderField ? renderField(element.elementName, element.options) : null;
+        })}
+        <div className={"button-container"}>
+          <button type={"submit"}>Submit</button>
+        </div>
```
Comment (thomascullen) [inline packages/ui-components/src/Form/index.tsx:93]: No need for curly when passing static strings.

File: packages/types/uiComponents.ts [hunk 2]
```diff
@@ -167,3 +173,14 @@ export interface RevealFormat {
...
   replace: string;
 }
+
+export interface FormPayload {
+  value: string;
+  isComplete: boolean;
+}
+
+export interface FormFrameClientMessages extends EvervaultFrameClientMessages {
+  EV_CHANGE: FormPayload;
```
Comment (thomascullen) [inline packages/types/uiComponents.ts:183]: We don't seem to ever fire this event. Can we remove it?

**Review by thomascullen — COMMENTED** [review]
Few quick things!

File: packages/types/uiComponents.ts (hunk 2 above)
Comment (boilsquid) [reply packages/types/uiComponents.ts:183]: For sure

**Comment by boilsquid** [comment]
> Few quick things!

Donzo!

File: packages/browser/lib/ui/form.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  EvervaultFrameHostMessages,
+  FormOptions,
+  FormPayload,
+  FormFrameClientMessages,
+} from "types";
+
+interface FormEvents {
+  ready: () => void;
+  error: () => void;
+  change: (payload: FormPayload) => void;
```
Comment (thomascullen) [inline packages/browser/lib/ui/form.ts:15]: Can remove this now

File: packages/browser/lib/ui/form.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  complete: (payload: FormPayload) => void;
```
Comment (thomascullen) [inline packages/browser/lib/ui/form.ts:16]: Here we have `complete` but bellow we dispatch `submitted` instead. We should change this to 

```
submitted: () => void
```

File: packages/browser/lib/ui/form.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+export default class Form {
+  values: FormPayload = {
+    value: "",
+    isComplete: false,
+  };
```
Comment (thomascullen) [inline packages/browser/lib/ui/form.ts:23]: Can remove this I think

File: packages/types/uiComponents.ts [hunk 6]
```diff
@@ -167,3 +173,13 @@ export interface RevealFormat {
   regex: RegExp;
   replace: string;
 }
+
+export interface FormPayload {
+  value: string;
+  isComplete: boolean;
+}
+
```
Comment (thomascullen) [inline packages/types/uiComponents.ts:181]: We prob dont need this interface anymore.

**Review by thomascullen — COMMENTED** [review]
View more tiny bits! All working well though.

**Review by thomascullen — APPROVED** [review]

**Pull request merged**